services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -840,32 +838,22 @@ blockNonConsentingConnections uid = do
       status <- putConnectionInternal (BlockForMissingLHConsent userLegalhold othersToBlock)
       pure $ ["blocking users failed: " <> show (status, othersToBlock) | status /= status200]
 
-setTeamLegalholdWhitelisted :: Member LegalHoldStore r => TeamId -> Sem r ()
-setTeamLegalholdWhitelisted tid = LegalHoldData.setTeamLegalholdWhitelisted tid
-
-setTeamLegalholdWhitelistedH :: Member LegalHoldStore r => TeamId -> Sem r Response
-setTeamLegalholdWhitelistedH tid = do
-  empty <$ setTeamLegalholdWhitelisted tid
-
-unsetTeamLegalholdWhitelisted :: Member LegalHoldStore r => TeamId -> Sem r ()
-unsetTeamLegalholdWhitelisted tid = LegalHoldData.unsetTeamLegalholdWhitelisted tid
-
-unsetTeamLegalholdWhitelistedH :: Member LegalHoldStore r => TeamId -> Sem r Response
+unsetTeamLegalholdWhitelistedH :: Member LegalHoldStore r => TeamId -> Sem r ()
 unsetTeamLegalholdWhitelistedH tid = do
   () <-
     error
       "FUTUREWORK: if we remove entries from the list, that means removing an unknown \
       \number of LH devices as well, and possibly other things.  think this through \
       \before you enable the end-point."
-  setStatus status204 empty <$ unsetTeamLegalholdWhitelisted tid
+  LegalHoldData.unsetTeamLegalholdWhitelisted tid
 
-getTeamLegalholdWhitelistedH :: Member LegalHoldStore r => TeamId -> Sem r Response
+getTeamLegalholdWhitelistedH ::
+  (Member LegalHoldStore r, Member (Error LegalHoldError) r) =>
+  TeamId ->
+  Sem r ()
 getTeamLegalholdWhitelistedH tid = do
   lhEnabled <- LegalHoldData.isTeamLegalholdWhitelisted tid
-  pure $
-    if lhEnabled
-      then setStatus status200 empty
-      else setStatus status404 empty
+  unless lhEnabled $ throw LegalHoldNotEnabled -- NB: Should switch to ErrorS / MapError
```

Comment:
Instead of throwing an exception here, we can simply use `MultiVerb` with two responses.

Reply:
That eliminates another ...H function. :)